adafruit_fram.py

```diff
@@ -0,0 +1,273 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2018 Michael Schroeder
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_fram`
+====================================================
+
+CircuitPython/Python library to support the I2C and SPI FRAM Breakouts.
+
+* Author(s): Michael Schroeder
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+ * Adafruit `I2C Non-Volatile FRAM Breakout
+   <https://www.adafruit.com/product/1895>`_ (Product ID: 1895)
+ * Adafruit `SPI Non-Volatile FRAM Breakout
+   <https://www.adafruit.com/product/1897>`_ (Product ID: 1897)
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+ * Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+"""
+
+# imports
+from micropython import const
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_FRAM.git"
+
+_MAX_SIZE_I2C = const(32767)
+_MAX_SIZE_SPI = const(8191)
+
+_I2C_MANF_ID = const(0x0A)
+_I2C_PROD_ID = const(0x510)
+
+class FRAM:
+    """
+    Driver base for the FRAM Breakout.
+    """
+
+    def __init__(self, max_size, write_protect=False, wp_pin=None):
+        self._max_size = max_size
+        self._wp = write_protect
+        self._wraparound = False
+        if not wp_pin is None:
+            #import digitalio
+            self._wp_pin = wp_pin
+            # Make sure write_prot is set to output
+            self._wp_pin.switch_to_output()
+            self._wp_pin.value = self._wp
+        else:
+            self._wp_pin = wp_pin
+
+    @property
+    def max_size(self):
```

Comment:
Should this be the standard `__len__` instead? That way you can use range on it in a loop.